.github/workflows/release-auto.yml

```diff
@@ -0,0 +1,86 @@
+# This GitHub action can publish assets for release when a tag is created.
+# Currently its setup to run on any tag that matches the pattern "v*" (ie. v0.1.0).
+#
+# This uses an action (paultyng/ghaction-import-gpg) that assumes you set your
+# private key in the `GPG_PRIVATE_KEY` secret and passphrase in the `PASSPHRASE`
+# secret. If you would rather own your own GPG handling, please fork this action
+# or use an alternative one for key handling.
+#
+# You will need to pass the `--batch` flag to `gpg` in your signing step
+# in `goreleaser` to indicate this is being used in a non-interactive mode.
+#
+name: release-auto2
+
+on:
+  pull_request:
+    types: [closed]
+
+jobs:
+  auto-release:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.merged == true }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Unshallow fetch
+        run: git fetch --prune --unshallow
+
+      - name: Get release label
+        uses: actions-ecosystem/action-release-label@v1
+        id: release-label
+        if: ${{ github.event.pull_request.merged == true }}
+
+      - name: Get latest tag
+        uses: actions-ecosystem/action-get-latest-tag@v1
+        id: get-latest-tag
+        if: ${{ steps.release-label.outputs.level != null }}
+      
+      - run: |
+          echo "Latest tag: ${{ steps.get-latest-tag.outputs.tag }}"
+          echo "Release level: ${{ steps.release-label.outputs.level }}"
+
+      - name: Bump the version
+        uses: actions-ecosystem/action-bump-semver@v1
+        id: bump-semver
+        if: ${{ steps.release-label.outputs.level != null }}
+        with:
+          current_version: ${{ steps.get-latest-tag.outputs.tag }}
+          level: ${{ steps.release-label.outputs.level }}
+
+      - run: |
+          echo "New version: ${{ steps.bump-semver.outputs.new_version }}"
+
+      - name: Push new tag
+        uses: actions-ecosystem/action-push-tag@v1
+        if: ${{ steps.release-label.outputs.level != null }}
+        with:
+          tag: ${{ steps.bump-semver.outputs.new_version }}
+          message: '${{ steps.bump-semver.outputs.new_version }}: PR #${{ github.event.pull_request.number }} ${{ github.event.pull_request.title }}'
+      
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        if: ${{ steps.release-label.outputs.level != null }}
+        with:
+          go-version: 1.18.x
+
+      - name: Import GPG key
+        id: import_gpg
+        uses: paultyng/ghaction-import-gpg@v2.1.0
+        if: ${{ steps.release-label.outputs.level != null }}
+        env:
+          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+          PASSPHRASE: ${{ secrets.PASSPHRASE }}
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v2
+        if: ${{ steps.release-label.outputs.level != null }}
+        with:
+          version: latest
+          args: release --rm-dist
```

Comment:
Looking at past runs, I see a deprecation warning for `--rm-dist`. It's recommended to use`--clean` instead